torch/csrc/distributed/c10d/CUDASymmetricMemory-inl.h

```diff
@@ -49,6 +52,15 @@ cas(uint32_t* addr, uint32_t compare, uint32_t val) {
   cuda::atomic_ref<uint32_t, cuda::thread_scope_system> ref(*addr);
   ref.compare_exchange_strong(compare, val, cuda::std::memory_order(Sem));
   return compare;
+#elif defined(USE_ROCM)
+  if (Sem == std::memory_order_acquire || Sem == std::memory_order_acq_rel) {
```

Comment:
```suggestion
  if constexpr (Sem == std::memory_order_acquire || Sem == std::memory_order_acq_rel) {
```

Reply:
curious can `__atomic_compare_exchange_n` with the necessary memory model be used here? I've seen `__atomic_load_n` with memmodel used in other places with ROCm.